scripts/entrypoint_openshift.sh

```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+export USER_NAME=geoserveruser
+export HOME=/home/geoserveruser
+if ! whoami &> /dev/null; then
+  if [ -w /etc/passwd ]; then
+    echo "${USER_NAME:-default}:x:$(id -u):0:${USER_NAME:-default} user:${HOME}:/sbin/nologin" >> /etc/passwd
+  fi
+fi
+cd $HOME
```

Comment:
The `cd` at the end was not used, I think. So we can omit it.

Reply:
> > Hello, the "root" group is because openshift runs containers with a random userID (so there is no way you can assign the files to the right user, you simply don't know in advance which userID will be used by openshift). The only certainty you have is that the group of the user will be root. This is documented in the guidelines doc mentioned in the PR.
> 
> We are not familiar with openshift, so we don't really know what the expected best practises are. However most of our docker image user use this image to transition from legacy monolithic setup into a microservice based deployment. In most cases, the Geoserver data directory already exists in the host file system, and bind mounted into the container, overlaying the dirpath inside the docker image. So it will definitely have different uid/gid. From what you describe, then there's no way to reconcile this case since we actually don't know what uid/gid the existing host folder will have.

normally this is solved using group permissions at openshift. i think it also sets the umask accordingly (but i didn't find a reference in the openshift docs).
that made me think, i should add the 10001 user back so that by default it runs as 10001 and is compatible with the current docker images in non-openshift context.


> In that case, can you suggest possible approach? Can we create an entrypoint scripts that automatically assign the folder permission to the correct groupt **at runtime**? Will that work? My hunch says it's not possible since the script inside already assigned with random UID which doesn't have power to change folder permission. But I maybe wrong.

that would not work as if the openshift user has no write access, nothing can be done at runtime to fix it ...

> > wrt the creation of directories: we didn't make any modification to the start.sh script, so are you sure there is a change wrt current behaviour ? i can do some more testing tomorrow. the modifications also don't do any user switching, permission denied seems strange. i will try to figure out what is going on tomorrow.
> 
> Sorry, to be clear, what @NyakudyaA did was testing the new image with **existing host folder** made by the previous image. This is just to test if the new image will break/not run if our user (people who uses the image) upgrade the image in their own environment/deployment. To summarize, it's not your script at fault. It's just that maybe the script doesn't expect this kind of usage, so we have to find a way to handle it gracefully. Which is, to correct the folder permission at runtime so it uses proper group.
> 
> Can you comment on my question @kervel @VBP8501 ?